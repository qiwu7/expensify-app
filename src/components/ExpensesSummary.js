import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral'; 
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export class ExpensesSummary extends React.Component {
    showMessage = () => {
        const numExpenses = this.props.numExpenses;
        const formatedTotalExpense = numeral(this.props.totalExpense / 100).format('$0,0.00');
        if (numExpenses === 0) {
            return 'No Expenses';
        } else {
            const expenseWord = numExpenses === 1 ? 'expense' : 'expenses';
            return <p>Viewing <span>{numExpenses}</span> {expenseWord} totalling <span>{formatedTotalExpense}</span></p>
        }
    };
    render() {
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">{this.showMessage()}</h1>
                    <div className="page-header__actions">
                        <Link className="button" to="/create">Add Expense</Link>
                    </div>
                </div>
            </div>
        );
    };
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        numExpenses: visibleExpenses.length,
        totalExpense: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);