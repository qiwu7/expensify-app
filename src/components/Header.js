import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => {
    return (
        <header>
            <h1>Expensify</h1>
            <NavLink to="/dashboard" activeClassName="is-active">Home</NavLink>
            <NavLink to="/create" activeClassName="is-active">Create</NavLink>
            <button onClick={props.startLogout}>Logout</button>
        </header>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    };
};

export default connect(undefined, mapDispatchToProps)(Header);