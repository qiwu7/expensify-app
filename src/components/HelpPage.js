import React from 'react';

const HelpPage = () => {
    return (
        <div>
            this is from my help component
        </div>
    );
};

export default HelpPage;