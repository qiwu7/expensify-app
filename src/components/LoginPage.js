import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = (props) => {
    return (
        <div>
            <h1>Login with gmail</h1>
            <button onClick={props.startLogin}>Login</button>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => dispatch(startLogin())
    };
};

export default connect(undefined, mapDispatchToProps)(LoginPage);