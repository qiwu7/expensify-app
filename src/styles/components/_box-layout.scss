.box-layout {
    align-items: center;
    background: url('/images/gg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 5px;
    color: rgb(153, 149, 149);
    padding: $l-size $m-size;
    text-align: center;
    width: 40rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}

.box-layout__subtitle {
    margin-bottom: $xl-size;
}