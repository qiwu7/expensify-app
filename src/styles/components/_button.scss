.button {
    background: $blue;
    border: none;
    border-radius: 3px;
    display: inline-block;
    color: white;
    font-size: $font-size-large;
    line-height: 1;
    font-weight: 300;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--remove {
    @extend .button;
    background: $light-grey;
}