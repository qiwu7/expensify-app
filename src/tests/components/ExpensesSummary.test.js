import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { filters } from '../fixtures/filters';

test('should render ExpensesSummary with no expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary numExpenses={0} totalExpense={0} />);
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpensesSummary with 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary numExpenses={1} totalExpense={9522} />);
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpensesSummary with 2 expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary numExpenses={2} totalExpense={952231324} />);
    expect(wrapper).toMatchSnapshot();
})