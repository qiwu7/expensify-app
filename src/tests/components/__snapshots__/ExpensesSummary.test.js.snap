// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      <p>
        Viewing 
        <span>
          1
        </span>
         
        expense
         totalling 
        <span>
          $95.22
        </span>
      </p>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with 2 expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      <p>
        Viewing 
        <span>
          2
        </span>
         
        expenses
         totalling 
        <span>
          $9,522,313.24
        </span>
      </p>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with no expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      No Expenses
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
